pkg/kubelet/kubelet.go

```diff
@@ -1267,13 +1268,27 @@ func (kl *Kubelet) SyncPods(pods []api.BoundPod) error {
 		pc := podContainer{podFullName, uid, containerName}
 		if _, ok := desiredContainers[pc]; !ok {
 			glog.V(1).Infof("Killing unwanted container %+v", pc)
-			err = kl.killContainer(container)
+			err = kl.killContainer(dockerContainers[ix])
 			if err != nil {
 				glog.Errorf("Error killing container %+v: %v", pc, err)
+			} else {
+				killed = append(killed, dockerContainers[ix].ID)
 			}
 		}
 	}
 
+	for {
+		running, err := dockertools.GetRunningContainers(kl.dockerClient, killed)
+		if err != nil {
+			glog.Errorf("Failed to poll container state: %v", err)
+		}
+		if len(running) == 0 {
+			break
+		}
+		glog.Infof("Some containers still running: %v", running)
+		time.Sleep(time.Second)
+	}
```

Comment:
When we stop a container, we set a timeout of 10s for the Docker stop call, which as I understand it sends a sigterm, followed by a sigkill 10s later.

However, I'm ok with trying and then looping as Dawn suggests.

The goroutines don't help, because we still need to wait for the containers to die, the kill command is async.